src/character.h

```diff
@@ -290,9 +291,34 @@ class Character : public Creature
         Character(Character &&) = default;
         Character &operator=(const Character &) = default;
         Character &operator=(Character &&) = default;
-        
+        struct trait_data : public JsonSerializer, public JsonDeserializer {
+            /** Key to select the mutation in the UI. */
+            char key = ' ';
```

Comment:
This belongs in the UI code, stashing it in the referent like this has caused a lot of pain over in the inventory code.